components/board/client.go

```diff
@@ -38,8 +38,6 @@ type client struct {
 
 type boardInfo struct {
 	name                  string
-	spiNames              []string
-	i2cNames              []string
 	analogReaderNames     []string
 	digitalInterruptNames []string
 	gpioPinNames          []string
```

Comment:
This wasn't on my list, but I suspect `gpioPinNames` also never gets written to! and on closer inspection, the `GPIOPinNames()` function is never used in non-test code and should also be removed.